cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -302,6 +302,30 @@ public void testMetricsConfigMap() {
         }
     }
 
+    @ParallelTest
+    public void testStrimziMetricsConfigs() {
+        Kafka kafkaAssembly = new KafkaBuilder(KAFKA)
+                .editSpec()
+                    .editKafka()
+                        .withNewStrimziMetricsReporterConfig()
+                            .withNewValues()
+                                .withAllowList("kafka_log.*", "kafka_network.*")
+                            .endValues()
+                        .endStrimziMetricsReporterConfig()
+                    .endKafka()
+                .endSpec()
+                .build();
+
+        List<KafkaPool> pools = NodePoolUtils.createKafkaPools(Reconciliation.DUMMY_RECONCILIATION, kafkaAssembly, List.of(POOL_CONTROLLERS, POOL_MIXED, POOL_BROKERS), Map.of(), KafkaVersionTestUtils.DEFAULT_KRAFT_VERSION_CHANGE, SHARED_ENV_PROVIDER);
+        KafkaCluster kafkaCluster = KafkaCluster.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafkaAssembly, pools, VERSIONS, KafkaVersionTestUtils.DEFAULT_KRAFT_VERSION_CHANGE, null, SHARED_ENV_PROVIDER);
+        String brokerConfig = kafkaCluster.generatePerBrokerConfiguration(1, ADVERTISED_HOSTNAMES, ADVERTISED_PORTS);
+
+        assertThat(brokerConfig, containsString("kafka.metrics.reporters=io.strimzi.kafka.metrics.YammerPrometheusMetricsReporter"));
+        assertThat(brokerConfig, containsString("prometheus.metrics.reporter.listener.enable=true"));
+        assertThat(brokerConfig, containsString("prometheus.metrics.reporter.listener=http://0.0.0.0:9404"));
+        assertThat(brokerConfig, containsString("allowlist=kafka_log.*,kafka_network.*"));
```

Comment:
Here you need to use `generatePerBrokerConfigurationConfigMaps` and check config maps content, see previous test.

Reply:
Refactored